src/hooks/useForm.ts

```diff
@@ -0,0 +1,21 @@
+import { useState } from "react";
+
+type UseFormReturn<T extends Record<string, any>> = {
+  form: T;
+  handleChange: (key: keyof T, value: any) => void;
+  setForm: React.Dispatch<React.SetStateAction<T>>;
+};
+
+export const useForm = <T extends Record<string, any>>(initialForm: T): UseFormReturn<T> => {
+  const [form, setForm] = useState<T>(initialForm);
+
+  const handleChange = (key: keyof T, value: any) => {
+    setForm((prev) => ({ ...prev, [key]: value }));
+  };
```

Comment:
안정성을 위해 타입 한정을 시키는게 좋을 듯 싶습니다! 어떠신가요?

```suggestion
type UseFormReturn<T extends Record<string, string | undefined>> = {
  form: T;
  handleChange: (key: keyof T, value: string | undefined) => void;
  setForm: React.Dispatch<React.SetStateAction<T>>;
};

export const useForm = <T extends Record<string, string | undefined>>(
  initialForm: T,
): UseFormReturn<T> => {
  const [form, setForm] = useState<T>(initialForm);

  const handleChange = (key: keyof T, value: string | undefined) => {
    setForm((prev) => ({ ...prev, [key]: value }));
  };

```

Reply:
저게 value에 뭐가 들어오던 확장 가능하게 any로 두긴 한건데 현재 기획에선 string | undefined만 쓰니 상관 없긴 하겠네여
제안해주신대로 업데이트 하겠습니다!